### DISCUSSION TITLE
Rename the Architect deployment target to just Lambda handler and take out Architect functionality

### DISCUSSION THREAD

**Discussion by candidia** [remix-run/remix#4981] [Proposals]
I wanted to use Serverless Framework to deploy my Remix app to AWS. It turns out that using the Architect deployment target is the easiest way to achieve this as I found [here](https://github.com/getlift/lift/discussions/288) because it generates the lambda handler we need. I think we should generalize this to just be an "AWS Lambda" deployment target instead of "Architect (AWS Lambda)" because this should be compatible with any IaC solution including Architect, Serverless Framework, CloudFormation, etc., all of which just need to reference the handler function path, without adding unnecessary dependencies like @architect/functions and the extra files like app.arc. Once you have the lambda handler and static assets, you can follow the docs in any IaC framework for referencing them.

There also seems to be concern about the Architect package enforcing HttpApiGateway and not allowing RestApiGateway, which is addressed in [another package](https://github.com/andersquist/serverless-remix). Perhaps this could also decouple that choice.

**Comment by meza** [comment]
We 100% need a way to get rid of Arc when the only thing we need is a lambda handler wrapper.

**Comment by simi-obs** [comment]
Is somebody looking at this? I do not have much experience with OSS contributing, but I would help, contribute to this.

So I suppose the base idea would be to create new package `remix-aws-lambda` exposing `createRequestHandler` function (similar to what we see from remix-architect). 

Returned handler would be capable to recognize and process events from:
- HTTP API GATEWAY
- REST API GATEWAY
- LAMBDA URL
- APPLICATION LOAD BALANCER
- CLOUDFRONT (Lambda@Edge)

and return correctly shaped responses.

Seems like a question of putting together solutions @CSFlorin posted (from SST, serverless-remix ....) .

Is this something worth looking at?

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
